bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Proposal.scala

```diff
@@ -115,6 +115,16 @@ case class Proposal(meta:Meta,
 
   def resetObservationMeta:Proposal = copy(observations = observations.map(_.copy(meta = None)))
 
+  // Calculate the program time as a ratio of total time.
+  def programTimeRatio: Double = {
+    def timeSum(extract: Observation => Option[TimeAmount]): TimeAmount =
+      TimeAmount.sum(observations.map(extract).flatten)
```

Comment:
Note that IDEA tells me I should use `flatMap` here, but as `Option` is not `GenTraversableOnce`, I can't.